runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -48,22 +48,37 @@ void J9::RecognizedCallTransformer::processIntrinsicFunction(TR::TreeTop* treeto
    TR::TransformUtil::removeTree(comp(), treetop);
    }
 
+
 bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
    {
+   static bool disableARMMaxMin = true; // (feGetEnv("TR_DisableARMMaxMin") != NULL);
+   static bool disableARMFMaxMin = true; // (feGetEnv("TR_DisableARMFMaxMin") != NULL);
+
    auto node = treetop->getNode()->getFirstChild();
    switch(node->getSymbol()->castToMethodSymbol()->getMandatoryRecognizedMethod())
       {
       case TR::java_lang_Integer_rotateLeft:
-      //case TR::java_lang_Math_abs_I: TODO OMR's iabs evaluator does not fully comply Java Specs
-      //case TR::java_lang_Math_abs_L: TODO OMR's labs evaluator does not fully comply Java Specs
+      case TR::java_lang_Math_abs_I:
+      case TR::java_lang_Math_abs_L:
       case TR::java_lang_Math_abs_F:
       case TR::java_lang_Math_abs_D:
-         return TR::Compiler->target.cpu.isX86();
+         return TR::Compiler->target.cpu.isX86() ||
+            TR::Compiler->target.cpu.isZ();
       case TR::java_lang_Math_max_I:
       case TR::java_lang_Math_min_I:
       case TR::java_lang_Math_max_L:
       case TR::java_lang_Math_min_L:
-         return TR::Compiler->target.cpu.isX86() && !comp()->getOption(TR_DisableMaxMinOptimization);
+         return (TR::Compiler->target.cpu.isX86() ||
+            TR::Compiler->target.cpu.isPower() ||
+            TR::Compiler->target.cpu.isZ() ||
+            TR::Compiler->target.cpu.isARM() && !disableARMMaxMin) &&
+            !comp()->getOption(TR_DisableMaxMinOptimization);
```

Comment:
Are we able to eliminate these platform checks and the `disableARMMaxMin` static variables by simply setting `TR_DisableMaxMinOptimization` on ARM? Similarly below.

Reply:
c03fdfd